packages/hooks/src/context.tsx

```diff
@@ -55,6 +55,7 @@ export function LiteflowProvider({
   const currentAddress = useMemo(() => {
     if (!authenticationToken) return null
     const res = decode<JwtPayload & { address: string }>(authenticationToken)
+    if (res.exp && res.exp < Math.ceil(Date.now() / 1000)) return null
```

Comment:
Looking at this a second time, it would make more sense to implement this verification in a custom `setAuthenticationToken` function.
This will both prevent to set an expired token and thus preventing to read the address from it.

If we want to be even more fancy, we could add a 1min "delay" to the expiration date, something like `Date.now() / 1000 - 60` to the token is still valid when the page load.

If we want to be super fancy, the expiration date should be checked every minute to prevent the case where the token is valid on first page load, but become expire after.

Reply:
There is a similar logic implemented in the starter-kit:
https://github.com/liteflow-labs/starter-kit/blob/672a8ab76fd052d03342170cb09918c15b842b6e/session.ts#L10-L18
Maybe we could move this JWT verification logic out of the starter-kit to be put in this package as an helper function?